crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/controller/CrownCourtContributionIntegrationTest.java

```diff
@@ -0,0 +1,230 @@
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.context.WebApplicationContext;
+import uk.gov.justice.laa.crime.contribution.CrownCourtContributionApplication;
+import uk.gov.justice.laa.crime.contribution.data.builder.TestModelDataBuilder;
+import uk.gov.justice.laa.crime.contribution.model.AppealContributionRequest;
+import uk.gov.justice.laa.crime.contribution.model.Assessment;
+import uk.gov.justice.laa.crime.contribution.model.Contribution;
+import uk.gov.justice.laa.crime.contribution.staticdata.enums.AssessmentStatus;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.httpBasic;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest(classes = CrownCourtContributionApplication.class, webEnvironment = DEFINED_PORT)
+@DirtiesContext
+public class CrownCourtContributionIntegrationTest {
+
+    private static final String CLIENT_CREDENTIALS = "client_credentials";
+    private static final String SCOPE_READ_WRITE = "READ_WRITE";
+    private static final String CLIENT_ID = "test-client";
+    private static final String CLIENT_SECRET = "secret";
+    private static final String LAA_TRANSACTION_ID = "999";
+    private static final String ENDPOINT_URL = "/api/internal/v1/contribution/appeal";
+    private static final String AUTH_URL = "/oauth2/token";
+
+    private MockMvc mvc;
+    private MockWebServer mockMaatApi;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private FilterChainProxy springSecurityFilterChain;
+
+    @BeforeAll
+    public void setUp() {
+        this.mvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext)
+                .addFilter(springSecurityFilterChain).build();
+    }
+
+    @BeforeEach
+    public void setupMaatApiServer() throws IOException {
+        mockMaatApi = new MockWebServer();
+        mockMaatApi.start(9999);
+
+        final Dispatcher dispatcher = new QueueDispatcher() {
```

Comment:
FYI if you find yourself having to add a number of different mock responses and error codes for different URL paths etc. consider the approach we took below. Although probably overkill for what you need right here and right now!

https://github.com/ministryofjustice/laa-crime-applications-adaptor/blob/ac7817e2c0bc4a5db39471ff96300098a2d354ef/crime-applications-adaptor/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/testutils/MockWebServerStubs.java